Controllers/DriverController.cs

```diff
@@ -0,0 +1,157 @@
+using advent_appointment_booking.Enums;
+using advent_appointment_booking.Models;
+using advent_appointment_booking.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using advent_appointment_booking.DTOs;
+
+namespace advent_appointment_booking.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DriverController : ControllerBase
+    {
+        private readonly IDriverService _driverService;
+
+        public DriverController(IDriverService driverService)
+        {
+            _driverService = driverService;
+        }
+
+        // Create a new driver
+        [HttpPost]
+        [Authorize(Policy = Policy.RequireTruckingCompanyRole)]
+        public async Task<IActionResult> CreateDriver([FromBody] Driver driver)
+        {
+            if (!ModelState.IsValid)
+            {
+                var errors = ModelState.Values.SelectMany(v => v.Errors).Select(e => e.ErrorMessage);
+                return BadRequest(new { message = "Validation failed", errors });
+            }
+
+            try
+            {
+                var result = await _driverService.CreateDriver(driver);
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                if (ex.Message.Contains("Trucking company doesn't exist"))
```

Comment:
Inside catch block, if else conditions are not required and no need to check exception messages like this